public/app/features/alerting/unified/components/rules/deleted-rules/DeletedRules.tsx

```diff
@@ -0,0 +1,175 @@
+        }
+        return <UpdatedByUser user={row.original.grafana_alert.updated_by} />;
+      },
+    },
+    {
+      id: 'title',
+      header: t('alerting.deletedRules.table.title', 'Title'),
+      disableGrow: true,
+      cell: ({ row }) => {
+        if (!rulerRuleType.grafana.rule(row.original)) {
+          return unknown;
+        }
+        return row.original.grafana_alert.title;
+      },
+    },
+    {
+      id: 'folder',
+      header: t('alerting.deletedRules.table.folder', 'Folder'),
+      disableGrow: true,
+      cell: ({ row }) => {
+        if (!rulerRuleType.grafana.rule(row.original)) {
+          return unknown;
+        }
+        return row.original.grafana_alert.namespace_uid;
+      },
+    },
+    {
+      id: 'group',
+      header: t('alerting.deletedRules.table.group', 'Group'),
+      disableGrow: true,
+      cell: ({ row }) => {
+        if (!rulerRuleType.grafana.rule(row.original)) {
+          return unknown;
+        }
+        return row.original.grafana_alert.rule_group;
+      },
+    },
+    {
+      id: 'created',
+      header: t('alerting.deletedRules.table.updated', 'Deletion Date'),
+      disableGrow: true,
+      cell: ({ row }) => {
+        if (!rulerRuleType.grafana.rule(row.original)) {
+          return unknown;
+        }
+        const value = row.original.grafana_alert.updated;
+        if (!value) {
+          return unknown;
+        }
+        return dateTimeFormat(value) + ' (' + dateTimeFormatTimeAgo(value) + ')';
+      },
+    },
+    {
+      id: 'actions',
+      disableGrow: true,
+      cell: ({ row }) => {
+        return (
+          <Stack direction="row" alignItems="center" justifyContent="flex-end">
+            <Button
+              variant="secondary"
+              size="sm"
+              icon="history"
+              onClick={() => {
+                showConfirmation(
+                  rulerRuleType.grafana.rule(row.original) ? getRowId(row.original.grafana_alert) : 'unknown'
+                );
+              }}
+            >
+              <Trans i18nKey="alerting.deletedRules.restore">Restore</Trans>
+            </Button>
+          </Stack>
+        );
+      },
+    },
+  ];
+
+  return (
+    <>
+      <InteractiveTable
+        pageSize={DELETED_RULES_PAGE_SIZE}
+        columns={columns}
+        data={deletedRules}
+        getRowId={(row) => {
+          if (!rulerRuleType.grafana.rule(row)) {
+            return 'unknown';
+          }
+          return getRowId(row.grafana_alert);
+        }}
+      />
+      <ConfirmRestoreDeletedRuleModal
+        ruleToRestore={restoreRule}
+        isOpen={confirmRestore}
+        onDismiss={hideConfirmation}
+        onRestoreSucess={() => trackDeletedRuleRestoreSuccess()}
+        onRestoreError={() => trackDeletedRuleRestoreFail()}
+      />
+    </>
+  );
+}
+
+function getRowId(row: GrafanaRuleDefinition) {
+  return `${row.namespace_uid}${row.namespace_uid}${row.title}${row.version}`;
```

Comment:
Is there `rule_uid` available for us? What happens if there were many rules with the same title?

Reply:
true!, yes, let's add the uid